### PR TITLE
Fix: output var validate keyAlreadyExists behavior.

### PR DESCRIPTION
**Pull request by aixgeek**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # 5625

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] TODO

# Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by aixgeek** [comment]
#5615 output var validate keyAlreadyExists. This behavior prevent input. when outputs have same prefix.

File: web/app/components/workflow/nodes/_base/components/variable/output-var-list.tsx [hunk 1]
```diff
@@ -64,6 +58,39 @@ const OutputVarList: FC<Props> = ({
...
+      const oldVar = list[index]
+      setEditVar(oldVar)
+    }
+  }, [list])
+
+  const handleVarNameBlur = useCallback((index: number) => {
+    return (e: React.ChangeEvent<HTMLInputElement>) => {
+      const oldKey = list[index].variable
+      const newKey = e.target.value
+      if (list.map(item => item.variable?.trim()).includes(newKey.trim())) {
```
Comment (zxhlyh) [inline web/app/components/workflow/nodes/_base/components/variable/output-var-list.tsx]: When the blur effect is applied, it keeps throwing errors, even when there is only one item.
Perhaps you should filter out its own item.

Comment (aixgeek) [reply web/app/components/workflow/nodes/_base/components/variable/output-var-list.tsx]: yes, I do it in next commit

fix: filter validate self key

**Comment by zxhlyh** [comment]
image
After clearing the variable type, selecting the type again will result in an error.

**Comment by aixgeek** [comment]
fixed

**Comment by aixgeek** [comment]
@zxhlyh dify outputs panel have two bugs:
1. when add new var, the new var maybe duplicate var name.
2. when change var name, the validate in change event can prevent normal input.

I have fixed these bugs.

**Comment by iamjoel** [comment]
@VincePotato Do you think it's necessary to auto rename the variable name when it's duplicate?

**Comment by aixgeek** [comment]
@iamjoel 
add a duplicate will cause a serious bug.
Steps to Reproduce：
1.add a var, it has same name
2.delete this var, it will delete the same var's type
3.can't choose var's type

image

**Comment by iamjoel** [comment]
> @iamjoel add a duplicate will cause a serious bug. Steps to Reproduce： 1.add a var, it has same name 2.delete this var, it will delete the same var's type 3.can't choose var's type
> 
> image

I'll fix the bug later.

**Comment by VincePotato** [comment]
> @VincePotato Do you think it's necessary to auto rename the variable name when it's duplicate?

I think we should notify and guide users to rename instead of automatically creating names, as automatically generated names usually lack distinctiveness.

**Comment by aixgeek** [comment]
@VincePotato @iamjoel  This pr is not rename the variable name. The major job is fix validate in change event. 

duplicate name problem is a extra work, it fix some problems.

**Comment by iamjoel** [comment]
> > @iamjoel add a duplicate will cause a serious bug. Steps to Reproduce： 1.add a var, it has same name 2.delete this var, it will delete the same var's type 3.can't choose var's type 添加重复项会导致严重的错误。重现步骤： 1.添加一个var，它具有相同的名称 2.删除这个var，它将删除相同的var类型 3.不能选择var的类型
> > image
> 
> I'll fix the bug later.稍后我会修复该错误。

Fixed with https://github.com/langgenius/dify/pull/5845/files

**Pull request closed without merging**